src/codegeneration/FromOptionsTransformer.js

```diff
@@ -57,18 +56,20 @@ import {options, transformOptions} from '../Options.js';
 export class FromOptionsTransformer extends MultiTransformer {
   /**
    * @param {ErrorReporter} reporter
-   * @param {UniqueIdentifierGenerator=} idGenerator
+   * @param {Options} options
    */
-  constructor(reporter, idGenerator = new UniqueIdentifierGenerator()) {
+  constructor(reporter, options) {
     super(reporter, options.validate);
+    var idGenerator = new UniqueIdentifierGenerator();
 
     var append = (transformer) => {
       this.append((tree) => {
-        return new transformer(idGenerator, reporter).transformAny(tree);
+        return new transformer(idGenerator, reporter, options).
+            transformAny(tree);
       });
     };
 
-    if (transformOptions.blockBinding) {
+    if (options.transformView('blockBinding')) {
```

Comment:
But `transformOptions` operations on the global `options` object and here we are removing the global object.

I suppose we could create a new local proxy object from the local reference to the `options` argument.  Is that what you meant by `options.transformOptions`?